### PR TITLE
#18363 pagination fix

### PR DESCRIPTION
**Pull request by jdotcms**
The pagination wasn't working as expected

### CODE REVIEW

**Review by nollymar — CHANGES_REQUESTED** [review]
A test should be implemented

**Review by freddyucv — APPROVED** [review]

**Review by erickgonzalez — APPROVED** [review]

**Comment by jdotcms** [comment]
hi @nollymar this already have unit test and postman too

**Comment by nollymar** [comment]
> hi @nollymar it already has unit test and postman too

If so, the test is not implemented correctly because it wasn't failing before this change

File: dotCMS/src/integration-test/java/com/dotcms/browser/BrowserAPITest.java [hunk 1]
```diff
@@ -113,7 +117,98 @@ public static void prepare() throws Exception {
...
+                .withHostOrFolderId(folder.getIdentifier())
+                .offset(0)
+                .showFiles(true)
+                .showFolders(true)
+                .showWorking(true)
+                .maxResults(10)
+                .build());
+
+        assertNotNull(resultMap);
+        assertEquals(100l, resultMap.get("total"));
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/browser/BrowserAPITest.java:149]: Codacy Issue found: [The String literal "total" appears 5 times in this file; the first occurrence is on line 149](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5476854)

File: dotCMS/src/main/java/com/dotcms/browser/BrowserAPIImpl.java [hunk 2]
```diff
@@ -60,6 +62,7 @@
     @CloseDBIfOpened
     public Map<String, Object> getFolderContent(final BrowserQuery browserQuery) throws DotSecurityException, DotDataException {
 
+        final MutableInt countItems                     = new MutableInt(0);
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/browser/BrowserAPIImpl.java:65]: Codacy Issue found: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5476854)

File: dotCMS/src/integration-test/java/com/dotcms/browser/BrowserAPITest.java [hunk 3]
```diff
@@ -113,7 +117,98 @@ public static void prepare() throws Exception {
...
+
+        List<Map<String, Object>> results = (List<Map<String, Object>>)resultMap.get("list");
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/browser/BrowserAPITest.java:151]: Codacy Issue found: [The String literal "list" appears 6 times in this file; the first occurrence is on line 151](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5476854)

**Review by dsilvam — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/browser/BrowserAPIImpl.java (hunk 2 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/browser/BrowserAPIImpl.java:65]: Codacy Issue found: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5476854)

File: dotCMS/src/integration-test/java/com/dotcms/browser/BrowserAPITest.java [hunk 4]
```diff
@@ -113,7 +117,108 @@ public static void prepare() throws Exception {
...
+                .withHostOrFolderId(folder.getIdentifier())
+                .offset(0)
+                .showFiles(true)
+                .showFolders(true)
+                .showWorking(true)
+                .maxResults(10)
+                .build());
+
+        assertNotNull(resultMap);
+        assertEquals(100l, resultMap.get("total"));
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/browser/BrowserAPITest.java:159]: Codacy Issue found: [The String literal "total" appears 5 times in this file; the first occurrence is on line 149](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5476854)

File: dotCMS/src/integration-test/java/com/dotcms/browser/BrowserAPITest.java [hunk 5]
```diff
@@ -113,7 +117,108 @@ public static void prepare() throws Exception {
...
+
+        List<Map<String, Object>> results = (List<Map<String, Object>>)resultMap.get("list");
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/browser/BrowserAPITest.java:161]: Codacy Issue found: [The String literal "list" appears 6 times in this file; the first occurrence is on line 151](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5476854)

**Review by nollymar — APPROVED** [review]

**Pull request merged**